components/Buttons/src/MDCButton.m

```diff
@@ -860,7 +861,7 @@ - (void)updateAlphaAndBackgroundColorAnimated:(BOOL)animated {
 }
 
 - (void)updateBackgroundColor {
-  super.backgroundColor = self.currentBackgroundColor;
+  self.layer.shapedBackgroundColor = self.currentBackgroundColor;
```

Comment:
Should we still be setting super.backgroundColor here? I'm surprised we were doing this in a method other than setBackgroundColor.

Reply:
I don't see why we would need to call super in this case, but please correct me if i'm unaware of something. What self.layer.shapedBackgroundColor does is call the sublayer's colorfill if there is a shape set, or otherwise call the self.layer.backgroundColor (which has the same effect as calling self.backgroundColor just without calling the setter itself of MDCButton). Because MDCButton subclasses UIButton and this call is called inside setBackgroundColor then i feel like the same flow remains.

Reply:
Ah got it! I can't tell from this PR whether the call to super is important, but if it looks like that behavior is expected here it might help to document that self's backgroundColor will eventually get set as expected.

Reply:
added a comment to clarify behavior. yeah i thought the same, i had looked at the commit history though and it looks like super is called instead of self because the setBackgroundColor method was deprecated and therefore it would of been a no-op if we had called self.backgroundColor: https://github.com/material-components/material-components-ios/commit/c99b5d0014eee336949165b473b6202eb0e8eea6